src/AppToolbar.jsx

```diff
@@ -1,22 +1,55 @@
 import React, { useEffect, useRef } from 'react'
-import { useActor } from '@xstate/react'
+import { useSelector } from '@xstate/react'
 import { AppBar, Icon, IconButton, Toolbar, Typography } from '@mui/material'
 import { toggleIconDataUri } from 'itk-viewer-icons'
 import toggleUICollapsed from './toggleUICollapsed'
 import './Panel.css'
 
 function AppToolbar(props) {
   const { service } = props
+  const send = service.send
   const collapseUIButton = useRef(null)
-  const [state, send] = useActor(service)
+  let contextCollapseUIButton = useSelector(
+    service,
+    (state) => state.context.collapseUIButton
+  )
+  const uiContainer = useSelector(service, (state) => state.context.uiContainer)
+  const uiCollapsed = useSelector(service, (state) => state.context.uiCollapsed)
+
+  const selectedName = useSelector(
+    service,
+    (state) => state.context.images.selectedName
+  )
+
+  const use2D = useSelector(service, (state) => state.context.use2D)
+
+  const planeUIGroup = useSelector(
+    service,
+    (state) => state.context.planeUIGroup
+  )
+
+  const viewMode = useSelector(service, (state) => state.context.viewMode)
+  const mainPlaneUIGroup = useSelector(
+    service,
+    (state) => state.context.main.planeUIGroup
+  )
 
   useEffect(() => {
-    state.context.main.collapseUIButton = collapseUIButton.current
+    contextCollapseUIButton = collapseUIButton.current
```

Comment:
I wonder if this does nothing due to "pass by value."   So our component simply has a local copy of `contextCollapseUIButton` reference.